src/overview.md

```diff
@@ -68,10 +68,12 @@ we'll talk about that later.
 - We then take the AST and [convert it to High-Level Intermediate
   Representation (HIR)][hir]. This is a compiler-friendly representation of the
   AST.  This involves a lot of desugaring of things like loops and `async fn`.
-- We use the HIR to do [type inference]. This is the process of automatic
-  detection of the type of an expression.
-- **TODO: Maybe some other things are done here? I think initial type checking
-  happens here? And trait solving?**
+- We use the HIR to do [type inference] (the process of automatic
+  detection of the type of an expression), [trait solving] (the process
+  of pairing up an impl with each reference to a trait), and [type
+  checking] (the process of converting the types found in the HIR
+  (`hir::Ty`), which represent the syntactic things that the user wrote,
+  into the internal representation used by the compiler (`Ty<'tcx>`)).
```

Comment:
```suggestion
  into the internal representation used by the compiler (`Ty<'tcx>`), and use that information to
  verify the type safety of a program based on analysis of a program's text).
```

Or something like that, feel free to suggest something better :)

Reply:
Seems good to me, except the `use` should be `using` :slightly_smiling_face: